kp_test.go

```diff
@@ -4145,3 +4145,199 @@ func TestSyncAssociatedResourcesError(t *testing.T) {
+				"state": 1,
+				"crn": "crn:v1:bluemix:public:kms:us-south:a/acount-id:some-nice-instance-id:key:key:12ka5-12ka5-12ka5-12ka5",
+				"deleted": false,
+				"dualAuthDelete": {
+				  "enabled": false
+				}
+			}
+		]
+	}`)
+
+	gock.New("http://example.com").
+		Get("/api/v2/keys").
+		MatchParams(map[string]string{"sort": "creationDate,state"}).
+		Reply(200).
+		Body(bytes.NewReader(listKeyResponse))
+
+	c, _, err := NewTestClient(t, nil)
+	gock.InterceptClient(&c.HttpClient)
+	defer gock.RestoreClient(&c.HttpClient)
+	c.tokenSource = &FakeTokenSource{}
+
+	sortStr, _ := GetKeySortStr(WithCreationDate(), WithState())
+
+	listKeysOptions := &ListKeysOptions{
+		Sort: sortStr,
+	}
+	keys, err := c.ListKeys(context.Background(), listKeysOptions)
+	assert.NoError(t, err)
+	assert.NotNil(t, keys)
+
+	/* 	Case 2 : When user wants to sort the keys in descending order based on last rotated date */
+	listKeyResponse = []byte(`{
+		"metadata":
+		{
+		  "collectionType": "application/vnd.ibm.kms.key+json",
+		  "collectionTotal": 2
+		},
+		"resources": [
+			{
+				"id": "12ka4-12ka4-12ka4-12ka4",
+				"name": "Key01June",
+				"type": "application/vnd.ibm.kms.key+json",
+				"aliases": [
+                    "alias01",
+                    "alias1"
+				],
+				"algorithmType": "AES",
+				"createdBy": "IBMid-55xxxxx",
+				"creationDate": "2022-06-01T11:27:40Z",
+				"lastUpdateDate": "2022-06-20T11:28:27Z",
+				"lastRotateDate": "2022-06-18T06:34:10Z",
+				"keyVersion": {
+				"id": "42ka1-42ka1-42ka1-42ka1",
+				"creationDate": "2022-06-18T06:34:10Z"
+				},
+				"keyRingID": "default",
+				"extractable": true,
+				"state": 2,
+				"crn": "crn:v1:bluemix:public:kms:us-south:a/acount-id:some-nice-instance-id:key:key:12ka5-12ka5-12ka5-12ka5",
+				"deleted": false,
+				"dualAuthDelete": {
+				"enabled": false
+				}
+			},
+			{
+				"id": "12ka4-12ka4-12ka4-12ka5",
+				"name": "Key2",
+				"type": "application/vnd.ibm.kms.key+json",
+				"algorithmType": "AES",
+				"createdBy": "IBMid-55xxxxx",
+				"creationDate": "2022-06-06T08:30:34Z",
+				"lastUpdateDate": "2022-06-12T14:19:24Z",
+				"lastRotateDate": "2022-06-12T14:19:24Z",
+				"keyVersion": {
+				  "id": "42ka1-42ka1-42ka1-42ka2",
+				  "creationDate": "2022-06-12T14:19:24Z"
+				},
+				"keyRingID": "default",
+				"extractable": true,
+				"state": 1,
+				"crn": "crn:v1:bluemix:public:kms:us-south:a/acount-id:some-nice-instance-id:key:key:12ka3-12ka3-12ka3-12ka5",
+				"deleted": false,
+				"dualAuthDelete": {
+				  "enabled": false
+				}
+			}
+		]
+	}`)
+
+	gock.New("http://example.com").
+		Get("/api/v2/keys").
+		MatchParams(map[string]string{"sort": "-lastRotateDate"}).
+		Reply(200).
+		Body(bytes.NewReader(listKeyResponse))
+
+	sortStr, _ = GetKeySortStr(WithLastRotateDateDesc())
+	listKeysOptions = &ListKeysOptions{
+		Sort: sortStr,
+	}
+
+	keys, err = c.ListKeys(context.Background(), listKeysOptions)
+	assert.NoError(t, err)
+	assert.NotNil(t, keys)
+
```

Comment:
Need to ensure that the correct HTTP call was actually made in both tests inside the function here

```suggestion
assert.True(t, gock.IsDone(), "Expected HTTP requests not called")
```
```